### ISSUE TITLE
State updates and rendering

### ISSUE THREAD

**Issue by DanLide** [L-U-C-I-F-E-R/Course-from-fullstackopen.com#3]
https://github.com/L-U-C-I-F-E-R/Course-from-fullstackopen.com/blob/b54e0a3378c5cd75ae9e91cc010ae7e52a728ae8/course/src/App.js#L57-L60

### TL;DR

You use `setInput` to update the `input` value each time the user types some symbol. Everything ok on this line.

On the next line, you try to log the "updated" `input` to the console. But actually, `input` will be updated only after the `HandlerEvent` finishes its work.

### Why it works so?

When you update the state (by calling `setInput`), React understands that UI should be re-rendered. 

So, when your handler will finish executing, React collect the updated state and re-renders the component. 
After this, you will see the updated state of input on the HTML page.

The main problem here is that you're using the `input` state from the previous render. Since you just passed the `event.target.value` to `setInput`, you can log `event.target.value` as well to see the last updated value:

```javascript
console.log(event.target.value)
```

In case you want to display an updated `input` state, you can just put it as a JSX element:

```javascript
return (
  ...
  <div>{input}</div>
  ...
);
```

And in case you want to log the updated input to the console, [useEffect](https://reactjs.org/docs/hooks-effect.html) can be helpful:

```javascript
useEffect(() => {
  console.log(input);
}, [input]); // Only re-run the effect if input changes
```

Hope that helps!
